libc/exit/exit.c

```diff
@@ -0,0 +1,1236 @@
+			TEST_ASSERT_EQUAL_INT(123, buf);
+		}
+
+		close(fd);
+		remove(TEST_EXIT_PATH);
+	}
+#endif
+}
+
+
+TEST_GROUP_RUNNER(unistd_exit)
+{
+	test_common.test_exitPtr = _exit;
+
+	RUN_TEST_CASE(unistd_exit, status_vals);
+	RUN_TEST_CASE(unistd_exit, exit_status_waitpid);
+	RUN_TEST_CASE(unistd_exit, chk_if_exits);
+	RUN_TEST_CASE(unistd_exit, unblock_thread_wait);
+	RUN_TEST_CASE(unistd_exit, unblock_thread_waitpid);
+	RUN_TEST_CASE(unistd_exit, close_streams);
+	RUN_TEST_CASE(unistd_exit, orphaned_child);
+	RUN_TEST_CASE(unistd_exit, new_parent_id);
+	RUN_TEST_CASE(unistd_exit, SIGCHLD_sent);
+	RUN_TEST_CASE(unistd_exit, per_thread_data_destructors);
+	RUN_TEST_CASE(unistd_exit, no_atexit);
+	RUN_TEST_CASE(unistd_exit, no_flush);
+	RUN_TEST_CASE(unistd_exit, no_handler);
+}
+
+
+TEST_GROUP(unistd_Exit);
+/* Changing test group name to unistd_Exit to make error messages more readable */
+CHANGE_TEST_GROUP(unistd_Exit, unistd_exit, status_vals)
+CHANGE_TEST_GROUP(unistd_Exit, unistd_exit, exit_status_waitpid);
+CHANGE_TEST_GROUP(unistd_Exit, unistd_exit, chk_if_exits);
+CHANGE_TEST_GROUP(unistd_Exit, unistd_exit, unblock_thread_wait);
+CHANGE_TEST_GROUP(unistd_Exit, unistd_exit, unblock_thread_waitpid);
+CHANGE_TEST_GROUP(unistd_Exit, unistd_exit, close_streams);
+CHANGE_TEST_GROUP(unistd_Exit, unistd_exit, orphaned_child);
+CHANGE_TEST_GROUP(unistd_Exit, unistd_exit, new_parent_id);
+CHANGE_TEST_GROUP(unistd_Exit, unistd_exit, SIGCHLD_sent);
+CHANGE_TEST_GROUP(unistd_Exit, unistd_exit, per_thread_data_destructors);
+CHANGE_TEST_GROUP(unistd_Exit, unistd_exit, no_atexit);
+CHANGE_TEST_GROUP(unistd_Exit, unistd_exit, no_flush);
+CHANGE_TEST_GROUP(unistd_Exit, unistd_exit, no_handler);
+
+
+TEST_GROUP_RUNNER(unistd_Exit)
+{
+	test_common.test_exitPtr = _Exit;
+
+	RUN_TEST_CASE(unistd_Exit, status_vals);
+	RUN_TEST_CASE(unistd_Exit, exit_status_waitpid);
+	RUN_TEST_CASE(unistd_Exit, chk_if_exits);
+	RUN_TEST_CASE(unistd_Exit, unblock_thread_wait);
+	RUN_TEST_CASE(unistd_Exit, unblock_thread_waitpid);
+	RUN_TEST_CASE(unistd_Exit, close_streams);
+	RUN_TEST_CASE(unistd_Exit, orphaned_child);
+	RUN_TEST_CASE(unistd_Exit, new_parent_id);
+	RUN_TEST_CASE(unistd_Exit, SIGCHLD_sent);
+	RUN_TEST_CASE(unistd_Exit, per_thread_data_destructors);
+	RUN_TEST_CASE(unistd_Exit, no_atexit);
+	RUN_TEST_CASE(unistd_Exit, no_flush);
+	RUN_TEST_CASE(unistd_Exit, no_handler);
+}
+
+
+TEST_GROUP_RUNNER(stdlib_exit)
+{
+	RUN_TEST_CASE(stdlib_exit, stream_flush);
+	RUN_TEST_CASE(stdlib_exit, atexit_few_calls);
+	RUN_TEST_CASE(stdlib_exit, atexit_register_inside);
+	RUN_TEST_CASE(stdlib_exit, atexit_two_nodes);
+}
+
+
+void runner(void)
+{
+	RUN_TEST_GROUP(unistd_exit);
+	RUN_TEST_GROUP(unistd_Exit);
+	RUN_TEST_GROUP(stdlib_exit);
+}
+
+
+int main(int argc, char *argv[])
+{
+	const char *tiniPath = "/tini";
+
+	/* Add test with tini (init system) if test is run on host in container */
+	if (access(tiniPath, F_OK) == 0 && getenv("use_tini") == NULL) {
```

Comment:
IMHO more readable, `use_tini` not set sound's like we don't want to use it.

```suggestion
	if (access(tiniPath, F_OK) == 0 && getenv("tini_used") == NULL) {
```

Reply:
I think `use_tini` sound better.